snap/snapcraft.yaml

```diff
@@ -30,6 +30,9 @@ apps:
     command: usr/bin/gnome-text-editor
     desktop: usr/share/applications/org.gnome.TextEditor.desktop
     common-id: org.gnome.TextEditor.desktop
+    environment:
+      GTK_USE_PORTAL: "1"
```

Comment:
Isn't this already set by `gnome` extension?

https://github.com/snapcore/snapcraft/blob/main/snapcraft/extensions/gnome.py#L149

Reply:
Yes, but it seems that there is specific code inside gnome-text-editor to detect this, so I added both just in case a future change replaces this with the other.

Reply:
> Yes, but it seems that there is specific code inside gnome-text-editor to detect this, so I added both just in case a future change replaces this with the other.

I think it's more of the `Gtk.FileChooserNative` thing. Because `Gtk.FileDialog` fixes it. I tried one thing today, when I run apps with `GTK_USE_PORTAL=0` and `GTK_USE_PORTAL=''` apps, still uses portal. May be they just search for this environment variable, or it may be something else.

Reply:
It first checks if it is working inside a flatpak, and if not, it checks that environment variable. If any of those cases is TRUE, it sets a global variable, and it checks it in several parts of the code through `glib_should_use_portal()`  function:

![imagen](https://github.com/ubuntu/gnome-text-editor/assets/105285003/b25ef6b2-24bc-4085-9525-0ac26583d76b)

Reply:
So I think that, just in case...

Reply:
Got it.